src/app/features/auth/pages/registration/registration-page.component.ts

```diff
@@ -0,0 +1,55 @@
+import { ChangeDetectionStrategy, Component } from '@angular/core'
+import { CommonModule } from '@angular/common'
+import { RegisterFormComponent } from '../../components/register-form/register-form.component'
+import { Select, Store } from '@ngxs/store'
+import { Router } from '@angular/router'
+import { AuthState } from '@core/auth/state/auth-state.service'
+import { firstValueFrom, Observable } from 'rxjs'
+import { RegisterData } from '@core/auth/types/credentials'
+import to from 'await-to-js'
+import {
+  Register,
+  SetIsLoading,
+  ValidateEmail
+} from '@core/auth/state/auth.actions'
+
+@Component({
+  selector: 'registration',
+  standalone: true,
+  imports: [CommonModule, RegisterFormComponent],
+  templateUrl: './registration-page.component.html',
+  host: { class: 'flex justify-center items-center h-screen' },
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class RegistrationPageComponent {
+  @Select((state: { auth: AuthState }) => state.auth.loading)
```

Comment:
Why not use standard selectors as NGXS docs recommed?

Reply:
Do you mean this way
 public loading$!: Observable<boolean>

  constructor( private readonly _store: Store) {
    this.loading$ = this._store.select( (state: { auth: AuthState }) => state.auth.loading )
  }

This approach also was described in the NGXS guide
@Select((state: { auth: AuthState }) => state.auth.loading)
  public loading$!: Observable<boolean>

So I decided to use ngxs native tools to get a selector.